torch/ao/quantization/dynamic_tracing/utils.py

```diff
@@ -0,0 +1,598 @@
+
+SeenOp.__repr__ = seen_op_repr  # type: ignore[assignment]
+
+QTensorInfo = collections.namedtuple(
+    'QTensorInfo',
+    [
+        'id',  # tensor ID
+        'inf_dtype',  # dtype at inference
+    ],
+)
+
+def op_needs_quantization(op: Callable) -> bool:
+    if op in functions_supported_by_quantization:
+        return True
+    if type(op) in module_types_supported_by_quantization:
+        return True
+    if op in q_mod_to_float_mod_mapping:
+        return True
+    return False
+
+# TODO: fix lint
+class ObserverWrapper(torch.nn.Identity):
+    def __init__(self, child):
+        super().__init__()
+        self.child = child
+
+def wrap_observers_in_placeholders(module: torch.nn.Module) -> None:
+    """
+    Wraps each child observer of `module` in a placeholder which prevents
+    the execution of the observer during the forward. This is useful to prevent
+    tracing the model with example inputs from contributing to calibration
+    statistics.
+    """
+    for name, child in module.named_children():
+        if isinstance(child, (ObserverBase, FakeQuantizeBase)):
+            wrapper = ObserverWrapper(child)
+            setattr(module, name, wrapper)
+        else:
+            wrap_observers_in_placeholders(child)
+
+def unwrap_observers_from_placeholders(module: torch.nn.Module) -> None:
+    """
+    Restores observers back to their original state.
+    """
+    # Note: we cannot use module.named_children() because we can
+    # have two different names refer to the same module, for example
+    # when we are reusing observers for torch.add scalar version.
+    for name, child in module._modules.items():
+        if child is None:
+            continue
+        if isinstance(child, ObserverWrapper):
+            unwrapped = child.child
+            setattr(module, name, unwrapped)
+        else:
+            unwrap_observers_from_placeholders(child)
+
+def trace_with_inputs(
+    model: torch.nn.Module,
+    example_args: Tuple[Any],
+) -> None:
+    with torch.no_grad():
+        old_training = model.training
+        model.eval()
+        wrap_observers_in_placeholders(model)
+        model(*example_args)
+        unwrap_observers_from_placeholders(model)
+        if old_training:
+            model.train()
+
+def pack_weights_for_functionals(
+    module: torch.nn.Module,
+) -> None:
+    """
+    Packs weights for functionals seen while tracing.
+    Note: weight packing for modules is handled by eager mode quantization
+    flow.
+    """
+    if hasattr(module, '_auto_quant_state'):
+        qstate = module._auto_quant_state
+        # find any ops which need packing
+        for idx, seen_op in qstate.idx_to_seen_ops.items():  # type: ignore[union-attr, operator]
+            packable_args_len = len(seen_op.packable_tensor_idx_to_name) + \
+                len(seen_op.packable_nontensor_idx_to_arg)
+            if packable_args_len == 0:
+                continue
+
+            if seen_op.type == F.conv2d:
```

Comment:
are we missing the other conv operators like 1d and 3d and corresponding ConvTranspose ones?

Reply:
yes, a lot of other operators are missing as well.  Future PRs could add support for all of them, to keep this PR of a manageable size.